From ac665343fc73ca54fe010fb4ca78c4e2f5134ffb Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:14:02 -0400
Subject: [alsa] usb-audio: Fix Creative VF0420 rate

Message-id: <140972823573-37-patchset-jkysela@redhat.com>
Patchwork-id: 90737
O-Subject: [RHEL 7.1 PATCH 036/442] ALSA: usb-audio: Fix Creative VF0420 rate
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Creative Live! Cam Vista IM (VF0420) reports rate of 16kHz while working
at 8kHz. The patch adds its USB ID to the existing quirk.

Signed-off-by: Pavel Hofman <pavel.hofman@ivitera.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 8c4b79cf216f467faacdfa7c6c40b6b85cc05782)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/format.c b/sound/usb/format.c
index 3525231..d244fd3 100644
--- a/sound/usb/format.c
+++ b/sound/usb/format.c
@@ -189,8 +189,10 @@ static int parse_audio_format_rates_v1(struct snd_usb_audio *chip, struct audiof
         chip->usb_id == USB_ID(0x0ccd, 0x00b1)) &&
        fp->altsetting == 5 && fp->maxpacksize == 392)
     rate = 96000;
-   /* Creative VF0470 Live Cam reports 16 kHz instead of 8kHz */
-   if (rate == 16000 && chip->usb_id == USB_ID(0x041e, 0x4068))
+   /* Creative VF0420/VF0470 Live Cams report 16 kHz instead of 8kHz */
+   if (rate == 16000 &&
+       (chip->usb_id == USB_ID(0x041e, 0x4064) ||
+        chip->usb_id == USB_ID(0x041e, 0x4068)))
     rate = 8000;
 
    fp->rate_table[fp->nr_rates] = rate;
-- 
1.7.1